Module.php

```diff
@@ -133,20 +140,40 @@ protected function installSchemaOrgVocabulary(ServiceLocatorInterface $serviceLo
     public function attachListeners(SharedEventManagerInterface $sharedEventManager)
     {
         $sharedEventManager->attach(
-            'Omeka\Entity\Item',
-            'entity.update.post',
-            function (Event $event) {
-                /** @var Item $item */
-                $item = $event->getTarget();
-                // check dataclass is catalag else reyrn
-                if ($item->getResourceClass()->getLabel() !== 'DataCatalog') {
-                    return;
-                }
-                /** @var \Omeka\Job\Dispatcher $dispatcher */
-                $dispatcher = $this->serviceLocator->get('Omeka\Job\Dispatcher');
-                //$dispatcher->dispatch(CreateCatalogDumpJob::class, $item->getId()); // async
-                $dispatcher->dispatch(CreateCatalogDumpJob::class, $item->getId(), $this->getServiceLocator()->get(\Omeka\Job\DispatchStrategy\Synchronous::class)); //sync
-            }
+            ItemAdapter::class,
+            'api.update.pre', // Do we need to get the pre or post events?
+            [$this, 'dispatchDumpJob']
```

Comment:
This is a prime candidate to delegate to a service in the application layer!

Reply:
I don't think that works. As I can see in all modules, this is how it is done...